src/group_theory/subgroup.lean

```diff
@@ -1186,6 +1186,15 @@ begin
   simp only [],
 end
 
+lemma le_ker_iff (f : G →* N) {H : subgroup G} : H ≤ f.ker ↔ H.map f = ⊥ :=
+begin
+  rw eq_bot_iff,
+  split,
+  { intros h x hx,
+    obtain ⟨y, hy1, rfl⟩ := hx,
+    exact h hy1 },
+  { exact λ h x hx, h ⟨x, hx, rfl⟩ },
+end
```

Comment:
We have this lemma for lie submodules as `lie_hom.map_bot_iff`. Should we call this lemma `monoid_hom.map_bot_iff` for consistency? Or does it makes more sense to rename the other one? cc @ocfnash

Reply:
To my ears, `map_eq_bot_iff` sounds better than `map_bot_iff`. However, `map_eq_bot_iff` is already taken, I think.

Reply:
I have strong feelings on naming here but I do think we should be consistent. In particular, if that means we rename `lie_hom.map_bot_iff` then that's fine by me.

I think we should also aim for consistency on the lemma statement. I note that `lie_hom.map_bot_iff` states `I.map f = ⊥ ↔ I ≤ f.ker` and thus has the LHS and RHS the opposite way round _and_ it's a `simp` lemma. That feels right to me (obviously!) but if it needs to change then so be it!

Reply:
I'm fine changing this lemma to `monoid_hom.map_bot_iff` or `monoid_hom.map_eq_bot_iff`, but then the existing `monoid_hom.map_bot_iff` should be renamed to something mentioning the injectivity assumption.

Reply:
To avoid confusion, the lemma being referred to is [`add_subgroup.map_eq_bot_iff : function.injective ⇑f → subgroup.map f H = ⊥ ↔ H = ⊥`](https://leanprover-community.github.io/mathlib_docs/group_theory/subgroup.html#add_subgroup.map_eq_bot_iff).

I think it might make sense to split this into two lemmas:
* `add_subgroup.map_eq_bot_iff : H.map f = ⊥ ↔ H ≤ f.ker` (and rename the lie one to `lie_ideal.map_eq_bot_iff`)
* `add_subgroup.ker_eq_bot : f.ker = ⊥ ↔ function.injective ⇑f`, since we already have `linear_map.ker_eq_bot`

Reply:
Ah, sorry about the confusion. Does my recent commit do what you're looking for?